package/merged/merged.go

```diff
@@ -47,9 +53,32 @@ func (f *FS) Open(path string) (fs.File, error) {
 		}
 		dirs = append(dirs, dir{file, stat})
 	}
+	// If we didn't find any files within the filesystem,
+	// return the missing file errors.
+	if len(dirs) == 0 {
+		return nil, notExists
+	}
 	return f.mergeDir(path, dirs)
 }
 
+// notExistsError is a collection of all errors while attempting to open a file
+// in one of the filesystems
+type notExists struct {
+	path   string
+	errors []error
+}
+
+// Error implements error and joins all the errors together
+func (n *notExists) Error() string {
+	return fmt.Errorf("merged: open %q. %w", n.path, errs.Join(n.errors...)).Error()
+}
+
+// This type of error should be an fs.ErrNotExist because all underlying errors
+// are also fs.ErrNotExists
+func (*notExists) Unwrap() error {
+	return fs.ErrNotExist
+}
```

Comment:
This is a surprisingly important fix. 

The error chain was getting lost when looking up a file that didn't exist in any filesystem (this can happen when there's errors). 

We now collect and continue the chain when we encounter `fs.ErrNotExist` errors. 

The collected errors themselves also satisfy `errors.Is(err, fs.ErrNotExist)`.